gradle-jdks/src/main/groovy/com/palantir/gradle/jdks/PalantirGradleJdkIdeaPlugin.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.gradle.jdks;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Files;
+import groovy.util.Node;
+import groovy.xml.XmlNodePrinter;
+import groovy.xml.XmlParser;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.util.function.Consumer;
+import javax.xml.parsers.ParserConfigurationException;
+import org.gradle.api.Plugin;
+import org.gradle.api.Project;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.plugins.ide.idea.model.IdeaModel;
+import org.xml.sax.SAXException;
+
+public class PalantirGradleJdkIdeaPlugin implements Plugin<Project> {
```

Comment:
Keep Java files in java source dirs (this is currently in groovy) otherwise they don't get errorprones. If this makes it difficult to access the groovy class you need (due to the silly source set ordering thing), move the groovy class to another project and depend on it, something like `gradle-jdks-groovy`.